src/Kestrel/ServerFactory.cs

```diff
@@ -36,7 +40,8 @@ public IDisposable Start(IServerInformation serverInformation, Func<IFeatureColl
         {
             var disposables = new List<IDisposable>();
             var information = (ServerInformation)serverInformation;
-            var engine = new KestrelEngine(_libraryManager);
+            var loggerFactory = _serviceProvider.GetRequiredService<ILoggerFactory>();
```

Comment:
I'm actually not certain if this'll work, but instead of using the service locator pattern (i.e. calling the service provider directly), can the `ILoggerFactory` be a parameter to the `ServerFactory` constructor and perhaps automatically resolved from DI?